src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,46 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
The `die` method uses `indexOf` and `splice` to remove the dead animal from the `Animal.alive` array, which is not optimal for performance. Consider using the `filter` method to remove dead animals from the array, as it avoids the need for nested loops and is generally more efficient.